File: youpinhui-manager-web/src/main/webapp/js/controller/itemCatController.js
 //Controller
app.controller('itemCatController' ,function($scope,$controller   ,itemCatService,typeTemplateService){	
	
	$controller('baseController',{$scope:$scope});//inheritance
	
    //test case , get all the data return a Json
	$scope.findAll=function(){
		itemCatService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//pagination
	$scope.findPage=function(page,rows){			
		itemCatService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//update the search result
			}			
		);
	}
	
	//find one object 
	$scope.findOne=function(id){				
		itemCatService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//save 
	$scope.save=function(){				
		var serviceObject; 				
		if($scope.entity.id!=null){//if it contains an id
		//	alert("update")
			serviceObject=itemCatService.update( $scope.entity ); //update   
		}else{
			$scope.entity.parentId=$scope.parentId;
		
			serviceObject=itemCatService.add( $scope.entity  );//add
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					 
		        	$scope.findByParentId($scope.parentId);//reload page
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//batch delete
	$scope.dele=function(){			
		//receive the selected id from checkbox 		
		//alert($scope.selectIds)
		itemCatService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.findByParentId($scope.parentId);//reload page
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//define a search entity
	
	//search
	$scope.search=function(page,rows){			
		itemCatService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//update the total result
			}			
		);
	}
	
    $scope.parentId=0;
	$scope.findByParentId=function(parentId){
	
		$scope.parentId=parentId;
		itemCatService.findByParentId(parentId).success(
			function(response){
				$scope.list=response;
			}
		)
	}

	$scope.level=1;
	$scope.setLevel=function(value){
		$scope.level=value;
	}

	$scope.changeCate=function(p_entity){
		if($scope.level==1){
			$scope.entity_1=null;
			$scope.entity_2=null;
		}
		if($scope.level==2){
			$scope.entity_1=p_entity;
			$scope.entity_2=null;
		}1
		if($scope.level==3){
			$scope.entity_2=p_entity;
		}

		$scope.findByParentId(p_entity.id);

	}

	$scope.options={type_template:[]};
	$scope.findTypeList =function(){
		
		typeTemplateService.selectOptionList().success(
			function(response){
				$scope.options={type_template:response};
				//alert($scope.options['type_template'])
				$("#typeTempList").select2({
					data:$scope.options['type_template']
				})
			}
		)
	}



});	

File: youpinhui-manager-web/src/main/java/com/youpinhui/manager/controller/ItemCatController.java
package com.youpinhui.manager.controller;

import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.youpinhui.entity.PageResult;
import com.youpinhui.entity.Result;
import com.youpinhui.pojo.TbItemCat;
import com.youpinhui.sellergoods.service.ItemCatService;


/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/itemCat")
public class ItemCatController {

	@Reference
	private ItemCatService itemCatService;
	
	/**
	 * return all 
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbItemCat> findAll(){			
		return itemCatService.findAll();
	}
	
	
	/**
	 * findall with pagination
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return itemCatService.findPage(page, rows);
	}
	
	/**
	 * Add
	 * @param itemCat
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbItemCat itemCat){
		try {
			System.out.println("inside item save!");
			itemCatService.add(itemCat);
			return new Result(true, " add success");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "add failed");
		}
	}
	
	/**
	 * Update
	 * @param itemCat
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbItemCat itemCat){
		try {
			itemCatService.update(itemCat);
			return new Result(true, "update success");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "update failed");
		}
	}	
	
	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbItemCat findOne(Long id){
		return itemCatService.findOne(id);		
	}
	
	/**
	 * batch delete
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try {
			itemCatService.delete(ids);
			return new Result(true, "delete success"); 
		} catch (Exception e) {
			e.printStackTrace(); 
			return new Result(false, "delete failed");
		}
	}
	
		/**
	 *  fuzzy search
	 * @param brand
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbItemCat itemCat, int page, int rows  ){
		return itemCatService.findPage(itemCat, page, rows);		
	}
	
	/**
	 * find parent by id
	 * @param brand
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/findByParentId")
	public List<TbItemCat> findByParentId(Long parentId  ){
		return itemCatService.findByParentId(parentId);		
	}
	
}
